src/sensor/ping360.cpp

```diff
@@ -58,7 +61,14 @@ void Ping360::startPreConfigurationProcess()
     msg.set_requested_id(CommonId::DEVICE_INFORMATION);
     msg.updateChecksum();
     writeMessage(msg);
-    _timeoutProfileMessage.start();
+
+    static int count = _ABRTotalNumberOfMessages;
+    if(count--) {
+        _preConfigurationTimer.start();
```

Comment:
Is this necessary? as the timer is not oneShot, it should fire every 100ms by itself, no?

Reply:
Yes. It should fire every 100ms, but this call restart the timer.
Imagine the case that we are receiving a new `CommonId::DEVICE_INFORMATION`, what we are doing here is:
- We have a valid message, **restart the time** to trigger the timeout procedure _if necessary_ and request for the message again to test the baudrate if necessary (by the count variable).